### DISCUSSION TITLE
Changing the bounds of the yaw optimisation

### DISCUSSION THREAD

**Discussion by henrysanders3** [NatLabRockies/floris#596] [Q&A] [answered]
Hi all,

I am having trouble altering the bounds of the yaw optimisation. I believe i have been through all of the input files and functions (e.g.yaw_optimizer_sr) to change the minimum and maximum yaw angles, however when I run the code it still optmises between 0 and 25 degrees. Where in the FLORIS files do i need to make changes to amend this?

Can the minimums and maximums be overwritten in the code?

Thanks in advance

**Comment by parkerdavid18** [comment]
Hi Henry,

I think I might be having the same problems as you, I am trying to alter the bounds to -25 to 25 degrees to analyse the optimal yaw angles. After altering all of the floris files it still optimise between -25 and 25 degrees. IS there a way this can be overwritten in the code?

Thanks

**Reply by henrysanders3** [reply]
@rafmudaf Do you know how to fix this? 

thanks

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi. Here's what the current `main` branch gives with the example `10_opt_yaw_single_ws.py`:
![image](https://user-images.githubusercontent.com/22119448/222396792-c49d9487-96f6-4bf2-8671-93f313bc6530.png)

Note that the default optimization bounds are 0 deg and +25 deg. The figure above shows optimal yaw angles between these limits. Then, if you want to change your optimization limits, you can change lines 43-44 to:
```
yaw_opt = YawOptimizationSR(fi, minimum_yaw_angle=-30.0, maximum_yaw_angle=30.0)
df_opt = yaw_opt.optimize()
```
and the result is:
![image](https://user-images.githubusercontent.com/22119448/222397055-223fa2f6-fba9-4746-9cba-1599fa9368cc.png)

Generally, you should never change the default choices in the FLORIS classes to change the options for one particular case. You should always do this on the script side, like above.

**Reply by henrysanders3** [reply]
@Bartdoekemeijer That works, thank you very much! Is there a way the rotor diameter can be overwritten in a similar way? Many thaks

**Reply by Bartdoekemeijer** [reply]
@henrysanders3 Let's keep that conversation in #595.
